File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        int[] ans = new int[2];
        for(int i=0; i<numbers.length; i++){
            
            int l = i+1;
            int r = numbers.length-1;
            int sec = target-numbers[i];
            while(r>=l){  
                int mid = l+(r-l)/2;
                
                if(numbers[mid] == sec){
                    ans[0] = i+1;
                    ans[1] = mid+1;
                    return ans;
                }
                else if(numbers[mid] > sec)
                    r = mid-1;
                else
                    l = mid+1;
            }
        }
        return ans;
    }
}

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        vector<int> ans;
            
        for(int i=0; i<numbers.size(); i++){
            int second = target-numbers[i];
        
            int left = i+1;
            int right = numbers.size()-1;
            while(right >= left){
                int mid = left+(right-left)/2;
            
                if(numbers[mid] == second){
                    ans.push_back(i+1);
                    ans.push_back(mid+1);
                    return ans;
                }else if(numbers[mid] > second){
                    right = mid-1;
                }else{
                    left = mid+1;
                }
            }
        }
        return ans;
    }
};